src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { lazy } from 'react'
+import { Button, ButtonVariation, Dialog } from '@harness/uicore'
+// eslint-disable-next-line no-restricted-imports
+import ChildComponentMounter from 'microfrontends/ChildComponentMounter'
+import { useStrings } from 'framework/strings'
+import { V1Agent } from 'services/gitops'
+
+// eslint-disable-next-line import/no-unresolved
+const CreateGitOpsAgent = lazy(() => import('gitopsui/CreateGitOpsAgent'))
+export interface CreateGitopsAgentModalProps {
+  toggleVisible: () => void
+  isVisible: boolean
+  onAgentCreated: (data: V1Agent) => void
+}
+
+interface GitopsAppProps {
+  provider?: V1Agent | null
+  onClose?: (data: any) => void
+  refetchOnSave?: (data: any) => void
+  // scope?: null
+  setNewAgent?: (data: V1Agent) => void
+}
+export default function CreateGitopsAgentModal({
+  isVisible,
+  onAgentCreated,
+  toggleVisible
+}: CreateGitopsAgentModalProps): JSX.Element {
+  const { getString } = useStrings()
+  const onAgentModalClose = (data: any): void => {
+    // console.log(data)
+  }
+  const refetchOnSave = (data: any): void => {
+    // console.log(data)
+  }
+  return isVisible ? (
+    <Dialog
+      onClose={toggleVisible}
+      isOpen={isVisible}
+      style={{
+        minWidth: 1175,
+        minHeight: 690,
+        borderLeft: 0,
+        padding: 0,
+        position: 'relative'
+      }}
+      enforceFocus={false}
+      usePortal
+      // portalClassName={css.dialog}
+      canOutsideClickClose={false}
+    >
+      <ChildComponentMounter<GitopsAppProps>
+        ChildComponent={CreateGitOpsAgent}
+        provider={null}
+        onClose={onAgentModalClose}
+        refetchOnSave={refetchOnSave}
```

Comment:
If implementation is not needed, we can remove the empty functions
```suggestion
        onClose={noop}
        refetchOnSave={noop}
```